llvm/test/tools/sycl-post-link/device-code-split/split-with-kernel-declarations.ll

```diff
@@ -6,7 +6,7 @@
 ; RUN: FileCheck %s -input-file=%t_0.sym --check-prefix CHECK-PER-SOURCE-SYM0
 ; RUN: FileCheck %s -input-file=%t_1.sym --check-prefix CHECK-PER-SOURCE-SYM1
 ;
-; RUN: sycl-module-split -split=source -S < %s -o %t1
+; RUN: llvm-split -sycl-split=source -S < %s -o %t1
```

Comment:
sycl-post-link tests should not call `sycl-module-split` or `llvm-split` tools. This is very confusing. 
@maksimsab, please, file an issue to refactor these tests.

Also, it's not clear why do we test pass functionality twice using different executables. Can we run just one tool to validate split logic?

Reply:
Migration of functionality of `sycl-post-link` to library hasn't been completed yet. For example, [ESIMD splitting](https://github.com/intel/llvm/blob/sycl/llvm/tools/sycl-post-link/sycl-post-link.cpp#L621) is going to be moved to the library during this sprint.
During the migration we keep testing of `sycl-post-link` since it is used in the default pipeline and we keep the testing of the library to be sure that there are no bugs.
In the end of the migration the library will be invoked from `clang-linker-wrapper` and `sycl-post-link` will be removed and these tests will move to the right directory. These tests weren't duplicated because it is easy to make a mistake when someone modifies one test and they forget to modify the equivalent duplicated test.

Proposed testing approach with `llvm-split` is the same as one that was used for AMD in https://github.com/llvm/llvm-project/pull/89245 .